### PR TITLE
refactor(batch): Use futures-async-stream to implement Projection executor

### PR DESCRIPTION
**Pull request by D2Lark**
## What's changed and what's your intention?

Implement Projection using Executor2 trait.

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/1941

### CODE REVIEW

**Comment by TennyZhuang** [comment]
bors try

**Comment by TennyZhuang** [comment]
bors seems failed, I will push it to another branch and merge it if CI passed.

File: src/batch/src/executor2/projection.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    identity: String,
+}
+
+impl Executor2 for ProjectionExecutor2 {
+    fn schema(&self) -> &Schema {
+        &self.schema
+    }
+
+    fn identity(&self) -> &str {
+        "TraceExecutor"
```
Comment (TennyZhuang) [inline src/batch/src/executor2/projection.rs:39]: identity is incorrect?

**Review by liurenjie1024 — APPROVED** [review]

**Comment by liurenjie1024** [comment]
bors try

**Comment by TennyZhuang** [comment]
triggered at https://github.com/singularity-data/risingwave/actions/runs/2214250176

**Comment by TennyZhuang** [comment]
CI passed, Thanks!

**Pull request merged**